caikit_nlp/modules/text_generation/fine_tuning.py

```diff
@@ -119,6 +120,23 @@ def train(
             shuffle=True,
         )
 
+        ### Dtype based processing
+        # NOTE: Following is not exhaustive list of all parameters
+        # for all dtypes
+        if torch_dtype == torch.float16:
```

Comment:
would it be possible to add small unit test(s) for these code paths or is the overhead for train/run pretty high right now even for unit tests?

Reply:
actually anything half precision is only available on GPUs, so the current unit tests setup won't work for this